python/Ganga/GPIDev/Credentials2/AfsToken.py

```diff
@@ -0,0 +1,122 @@
+from __future__ import absolute_import
+
+import datetime
+import os
+import re
+import subprocess
+from getpass import getpass
+from glob import glob
+
+import Ganga.Utility.logging
+from Ganga.Utility.Shell import Shell
+
+from .ICredentialInfo import ICredentialInfo, cache
+from .ICredentialRequirement import ICredentialRequirement
+from .exceptions import CredentialRenewalError
+
+logger = Ganga.Utility.logging.getLogger()
+
+info_pattern = re.compile(r"^User's \(AFS ID \d*\) tokens for (?P<id>\w*@\S*) \[Expires (?P<expires>.*)\]$", re.MULTILINE)
+
+
+class AfsTokenInfo(ICredentialInfo):
+    """
+    A wrapper around an AFS token
+
+    For now it is very CERN-specific (or at least only follows the CERN use-case)
+    """
+
+    def __init__(self, requirements, check_file=False, create=False):
```

Comment:
I think it's fair enough to expect people to read up on the base class otherwise it'll just be repeating what's there. And I would guess there would be links in the generated docs anyway.